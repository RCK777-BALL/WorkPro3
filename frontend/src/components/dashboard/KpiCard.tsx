/*
 * SPDX-License-Identifier: MIT
 */

import React from 'react';
import { ResponsiveContainer, LineChart, Line } from 'recharts';

export interface KpiCardProps {
  title: string;
  value: string | number;
  deltaPct?: number;
  series?: number[];
}

const KpiCard: React.FC<KpiCardProps> = ({ title, value, deltaPct, series = [] }) => {
  const deltaClass = deltaPct && deltaPct < 0 ? 'text-red-600' : 'text-green-600';

  const chartData = series.map((v, i) => ({ index: i, value: v }));

  return (
    <div className="rounded border bg-card p-4 text-card-foreground shadow">
      <div className="text-sm text-neutral-500">{title}</div>
      <div className="mt-1 text-2xl font-semibold">{value}</div>
      {typeof deltaPct === 'number' && (
        <div className={`text-xs ${deltaClass}`}>
          {deltaPct >= 0 ? '+' : ''}
          {deltaPct.toFixed(1)}%
        </div>
      )}
      {series.length > 0 && (
        <div className="mt-2 h-8 w-full">
          <ResponsiveContainer width="100%" height="100%">
            <LineChart data={chartData}>
              <Line
                type="monotone"
                dataKey="value"
                stroke="var(--color-brand-start)"
                dot={false}
                strokeWidth={2}
              />
            </LineChart>
          </ResponsiveContainer>
        </div>
      )}
    </div>
  );
};

export default KpiCard;
