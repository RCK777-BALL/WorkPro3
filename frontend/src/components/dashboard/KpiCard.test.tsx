/*
 * SPDX-License-Identifier: MIT
 */

import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import KpiCard from './KpiCard';

describe('KpiCard', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <KpiCard title="Test" value={100} deltaPct={5} series={[]} />,
    );
    expect(container).toMatchSnapshot();
  });
});
