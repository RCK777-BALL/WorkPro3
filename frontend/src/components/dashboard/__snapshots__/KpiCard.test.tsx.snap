exports[`KpiCard matches snapshot 1`] = `
<div>
  <div class="rounded border bg-card p-4 text-card-foreground shadow">
    <div class="text-sm text-neutral-500">
      Test
    </div>
    <div class="mt-1 text-2xl font-semibold">
      100
    </div>
    <div class="text-xs text-green-600">
      +5.0%
    </div>
  </div>
</div>
`;
